src/AccessControl/ZopeGuards.py

```diff
@@ -262,6 +265,31 @@ def __next__(self):
     next = __next__
 
 
+class _SafeMappingView:
+    __allow_access_to_unprotected_subobjects__ = 1
+
+    def __iter__(self):
+        for e in super().__iter__():
+            guard(self._mapping, e)
```

Comment:
This might not be correct when applied to keys:
the `guard` signature (`container, value, index=None`) suggests that it verifies that *value* can be accessed from *container*. For a mapping as *container*, it is strange that its keys could be treated in the same way as its values.
However, we pass the `name` parameter to `validate` as `None`. Potentially, this means that we in fact check that the value is publicly accessible. This would work for both mapping keys and values.

If you think the check is correct, I would prefer a corresponding comment.